src/Symfony/Component/Messenger/Transport/RedisExt/Connection.php

```diff
@@ -169,6 +170,41 @@ public function get(): ?array
             }
         }
 
+        try {
```

Comment:
as written in my comment this should be moved into a function which is called on first call of get and when a xpending messages reach maybe the claimTimeout.